src/backend/distributed/planner/distributed_planner.c

```diff
@@ -1467,6 +1470,29 @@ FinalizeNonRouterPlan(PlannedStmt *localPlan, DistributedPlan *distributedPlan,
 	/* add original range table list for access permission checks */
 	finalPlan->rtable = list_concat(finalPlan->rtable, localPlan->rtable);
 
+#if PG_VERSION_NUM >= PG_VERSION_16
+
+	/*
+	 * Original range table list is concatenated to final plan's range table list
+	 * therefore all the perminfoindexes should be updated to their value
+	 * PLUS the length of final plan's perminfos.
+	 */
+	int list_length_final_permInfos = list_length(finalPlan->permInfos);
```

Comment:
I think I noted in the other PR as well, the variable name `list_length_final_permInfos` is confusing as there are lots of lists, and final list maybe confusing.

maybe: final_plan_last_perminfo_index or such could be better?

Reply:
Yeah, you did. However I changed something here: previously I was concatenating perminfos right after this line. But now, I concatenate perminfos after the loop - this way it's readable.

However, I agree that a name that includes perminfoindex make this even more readable. So, will change it. Thanks.